tests/components/application-connector/docs/compass-runtime-agent-tests.md

```diff
@@ -0,0 +1,135 @@
+# Compass Runtime Agent
+
+**Table of Contents**
+
+- [Design and architecture](#design-and-architecture)
+- [Building](#building)
+- [Running](#running)
+  - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally)
+  - [Test setup](#test-setup---compass-runtime-agent-configuration)
+  - [Run the tests](#run-the-tests)
+- [Debugging](#debugging)
+  - [Running without cleanup](#running-without-cleanup)
+
+## Design and architecture
+
+The tests consist of:
+- [Test resources](../resources/charts/compass-runtime-agent-test/) used to perform the test
+- [Test runner](../test/application-connectivity-validator/) with all the test cases
+
+The tests are executed as a Kubernetes Job on a Kyma cluster where the tested Compass Runtime Agent is installed. The test Job is deployed in the `test` Namespace.
+
+![Compass Runtime Agent tests architecture](assets/compass-runtime-agent-tests-architecture.svg)
+
+The interactions between components are the following:
+
+1. Compass Runtime Agent periodically fetches certificates from Compass Connector.
+2. Compass Runtime Agent periodically fetches applications from Compass Director.
```

Comment:
```suggestion
2. Compass Runtime Agent periodically fetches Applications from Compass Director.
```
I think it should be `Applications`, correct?

Reply:
I don't think so. Here we meant the applications in Compass, so they're not the Kyma applications, although they do have their equivalents in Kyma. Am I right, @franpog859 ?

Reply:
Well, in Compass they are called Systems, in Kyma they are Applications but in a reality, they are just some combined specs of some APIs :really-confused-fractal-shrug: In my opinion you can go with whatever you want